packages/cli/src/utils.ts

```diff
@@ -149,63 +149,125 @@ export async function isYarnManager(): Promise<boolean> {
   return !!yarnPath;
 }
 
-// devDep: dep@version eg. typescript@4.4.4 or typescript or etc
-export async function bumpDevDep(devDep: string): Promise<void> {
+export async function isPnpmManager(): Promise<boolean> {
+  const pnpmPath = await findup('pnpm-lock.yaml', {
+    cwd: process.cwd(),
+  });
+
+  return !!pnpmPath;
+}
+
+export async function getLockfilePath(): Promise<string | null> {
+  const yarnPath = await findup('yarn.lock', {
+    cwd: process.cwd(),
+  });
+
+  if (yarnPath) {
+    return yarnPath;
+  }
+
+  const pnpmPath = await findup('pnpm-lock.yaml', {
+    cwd: process.cwd(),
+  });
+
+  if (pnpmPath) {
+    return pnpmPath;
+  }
+
+  const npmPath = await findup('package-lock.json', {
+    cwd: process.cwd(),
+  });
+
+  if (npmPath) {
+    return npmPath;
+  }
+
+  return null;
+}
+
+export async function getModuleManager(): Promise<'yarn' | 'npm' | 'pnpm'> {
   const isYarn = await isYarnManager();
-  // check if npm or yarn
-  const binAndArgs = {
-    bin: isYarn ? 'yarn' : 'npm',
-    args: isYarn
-      ? ['add', '-D', `${devDep}`, '--ignore-scripts']
-      : ['install', `${devDep}`, '--save-dev', '--ignore-scripts'],
-  };
+  const isPnpm = await isPnpmManager();
```

Comment:
You can use Promise.all like here:
https://github.com/rehearsal-js/action/blob/master/src/pm.ts

Reply:
oh nice. ok